src/lib/dma.c

```diff
@@ -189,10 +189,13 @@ void dma_sg_free(struct dma_sg_elem_array *elem_array)
 	dma_sg_init(elem_array);
 }
 
-void dma_buffer_copy_from(struct comp_buffer *source, uint32_t source_bytes,
-			  struct comp_buffer *sink, uint32_t sink_bytes,
+void dma_buffer_copy_from(struct comp_buffer *source, struct comp_buffer *sink,
 			  dma_process_func process, uint32_t samples)
 {
+	uint32_t source_bytes = audio_stream_sample_bytes(&source->stream) *
```

Comment:
Replacing specific dma buffer handling in host/dai components with consistent use of audio-stream api is a good idea since the dma buffer might be handled as an audio-stream by other parts as well (like pcm-converter code). 

Regarding this specific line, when called by dai component, the dd->frame_fmt used previously is replaced with source->stream.frame_fmt now, so assert(==) before a call might be a good idea to make sure they are consistently updated. Another option is to remove dd->frame_fmt completely and pass the setting to the audio-stream if possible (audio-stream already attached when dd->frame_fmt is configured?)

Reply:
can we make sure there are no dependencies on PCM formats in DMA buffers? we will use DMAs for compressed formats.

Reply:
>can we make sure there are no dependencies on PCM formats in DMA buffers? we will use DMAs for compressed formats.

I'm not familiar with compressed audio streams, but I guess it should be next data format, like `SOF_IPC_FRAME_MP3_LE`, isn't it? If so, then conversion should be done in pcm_converter, in flow like `host.dma_buffer->(mp3) pcm_converter-> (arbitrary pcm) host.local_buffer` or are you going to do separate component and create flow like `host.dma_buffer->(mp3) pcm_converter-> (mp3) host.local_buffer -> (mp3) decompresser -> (pcm, eg s32) pcm_converter -> (arbitrary pcm)` ?

Reply:
I don't think converters will really work on compressed formats. I think in the compress situation the pipeline must be something like `host.dma_buffer->(mp3) decompresser ->(pcm, say s24le) pcm_converter ->(pcm, whichever is demanded by further components)`

Reply:
Then passing number of bytes from source buffer in `dma_buffer_copy_*()` probably will be better idea - more universal solution.